main.js

```diff
@@ -0,0 +1,221 @@
+
+
+const form = document.querySelector("form");
+
+const currentWeather = document.querySelector("#current-weather")
+
+const today = document.querySelector(".today")
+
+const tomorrow = document.querySelector(".tomorrow")
+
+const dayAfter = document.querySelector(".day-after-tomorrow")
+
+const weatherWidget = document.createElement("img")
+
+const main = document.querySelector("main")
+
+const choose =  document.querySelector("#choose")
+
+const pSearch = document.querySelector("#pSearch")
+
+const previousData = []
+
+const ul = document.querySelector(".ulP")
+
+const h4result = document.querySelector('.h4_result')
+
+const convertForm = document.querySelector('.convertor-form')
+
+
+
+form.addEventListener("submit",(event)=>{
+  event.preventDefault();
+let location = event.target.location.value
+getWeatherApi(location)
+choose.style.display = "none"
+pSearch.style.display = "none"
+});
+
+convertForm.addEventListener("submit", (event) => {
+  event.preventDefault();
+  // class CelsiustoFarenheit
+const c = document.querySelector(".c").checked
+const f = document.querySelector(".f").checked
+if (c) {
+
+   h4result.textContent =  parseFloat( farenheitToCelcius(event.target.temp.value)).toFixed(2) + "°C"
+}
+if (f){
+  h4result.textContent = parseFloat(celciusToFarenheit(event.target.temp.value)).toFixed(2)+ " °F"
+}
+})
+
+function celciusToFarenheit(degree) {
+  return (degree * 9) / 5 + 32;
+}
+function farenheitToCelcius(degree) {
+  return ((degree - 32) * 5) / 9;
+}
+
+
+function addToPrevious(cityName,temp) {
+  const li = document.createElement("li")
+  li.innerHTML = `<a href="javascript: getWeatherApi('`+cityName+`')">`+ cityName +'  '+temp+ `F</a>`
+   if (!previousData.some(x => x === cityName))
+    {
+     ul.append(li)
+    }
+}
+
+function getWeatherApi(location){
+const deleteCity = document.querySelectorAll(".mainWeather")
+//this will delete the previous search
+if(deleteCity.length > 0){
+  for(let i = 0; i < deleteCity.length; i++){
+    deleteCity[i].remove()
+  }
+}
+
+  const BASE_URL = `https://wttr.in/${location}?format=j1`
+  fetch(BASE_URL)
+  .then((response) => response.json())
+  .then((result) => { 
+   getWeather(result)
+
```

Comment:
you are not clearing your form which is causing you to fail tests. Once you fix that you pass most of the tests.